src/modules/Extension/Service.php

```diff
@@ -438,11 +413,14 @@ public function uninstall(\Model_Extension $ext)
 
     public function downloadAndExtract($type, $id)
     {
-        $ext = $this->di['extension'];
-        $manifest = $ext->getExtension($id, $type);
+        $latest = $this->di['extension_directory']->getLatestExtensionRelease($id);
+
+        if (!isset($latest['download_url'])) {
+            throw new \Exception('Coudn\'t find a valid download URL for the extension.');
+        }
 
-        if (!isset($manifest['download_url'])) {
-            throw new \Exception('Invalid download URL for the extension');
+        if (!$this->di['extension_directory']->isExtensionCompatible($id)) {
+            throw new \Exception('This extension is not compatible with your version of FOSSBilling. Please update FOSSBilling to the latest version and try again.');
```

Comment:
Sorry, just realized another minor thing:
I'd personally prefer it if we used a placeholder and then displayed the minimum version required for the extension the user is trying to install, rather than telling them to update to the latest version